src/models/multitrackmodel.cpp

```diff
@@ -3087,39 +3088,48 @@ void MultitrackModel::moveTrack(int fromTrackIndex, int toTrackIndex)
         }
     }
 
-    m_tractor->remove_track(fromMltIndex);
-    m_tractor->insert_track(*producer, toMltIndex);
-
-    // remove_track() and insert_track() modify a & b tracks on the transitions.
-    // Here we will restore them to the right values
-    // Also move around the blend mode settings to follow the tracks that moved.
+    // Remove all the transitions
+    // (remove_track() and insert_track() would mess them up)
     service.reset(m_tractor->producer());
+    QScopedPointer<Mlt::Field> field(m_tractor->field());
     while (service && service->is_valid()) {
-        if (service->type() == mlt_service_transition_type) {
-            Mlt::Transition t((mlt_transition) service->get_service());
-            int aTrack = 0;
-            int bTrack = bTracks.front();
-            bTracks.pop_front();
-            if (service->get("mlt_service") != QString("mix") ) {
+        Mlt::Service s(service->get_service());
+        service.reset(service->producer());
+        if (s.type() == mlt_service_transition_type) {
+            field->disconnect_service(s);
+            s.disconnect_all_producers();
+        }
+    }
 
-                if (bTrack == 1) {
-                    t.set("disable", 1);
-                } else {
-                    // video transitions mix with track 1
-                    // (except track 1 which mixes with track 0)
-                    aTrack = 1;
-                    t.set("disable", 0);
-                }
-                // Transfer the blend mode to this video transition
-                if (!modes[bTrack].isEmpty()) {
-                    t.set("1", modes[bTrack].toUtf8().constData());
-                } else {
-                    t.clear("1");
+    // Remove the track and add it in the new position
+    m_tractor->remove_track(fromMltIndex);
+    m_tractor->insert_track(*producer, toMltIndex);
+
+    // Add the transitions back in the new positions
+    for (auto bTrack : audioTransitions.keys()) {
```

Comment:
`clazy` integration into Qt Creator gives me 2 warnings about this line:
> - [allocating an unneeded temporary container](https://github.com/KDE/clazy/blob/1.11/docs/checks/README-container-anti-pattern.md) (I still prefer a C++ range loop over using an iterator, and this is a short list of ints.)
> - [range loop might detach container](https://github.com/KDE/clazy/blob/1.11/docs/checks/README-range-loop-detach.md) (I personally tend not to use `qAsConst` to resolve this.)

Here is one approach that still makes a copy of the keys:
```
-    for (auto bTrack : audioTransitions.keys()) {
+    const auto keys = audioTransitions.keys();
+    for (auto &bTrack : keys) {
```
Alternatively using an iterator:
```
-    for (auto bTrack : audioTransitions.keys()) {
+    for (auto i = audioTransitions.keyBegin(); i != audioTransitions.keyEnd(); i++) {
+        int bTrack = *i;
```

Reply:
Done. I prefer the iterator.